### ISSUE TITLE
Expose node ports to host

### ISSUE THREAD

**Issue by durandom** [minishift/minishift#887]
I'm trying to expose a non-http port to my host (e.g. postgresql or in my case kafka 9092) and follow https://docs.openshift.org/latest/dev_guide/getting_traffic_into_cluster.html#using-externalIP

oc create -f loadbalancer.yml

```yaml
apiVersion: v1
kind: Service
metadata:
  name: kafka-ingress
spec:
  ports:
  - name: kafka
    port: 9092
  type: LoadBalancer
  selector:
    name: apache-kafka
```

creates this service:

```yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: 2017-05-11T14:43:47Z
  name: kafka-ingress
  namespace: myproject
  resourceVersion: "2110"
  selfLink: /api/v1/namespaces/myproject/services/kafka-ingress
  uid: 3d74c24c-3658-11e7-a52d-5add44d633b6
spec:
  clusterIP: 172.30.119.105
  deprecatedPublicIPs:
  - 172.29.8.80
  externalIPs:
  - 172.29.8.80
  ports:
  - name: kafka
    nodePort: 30528
    port: 9092
    protocol: TCP
    targetPort: 9092
  selector:
    name: apache-kafka
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - ip: 172.29.8.80
```

now how do I get to the ip, which seems to be local to the minishift vm

I'm on osx, xhyve

**Comment by hferentschik** [comment]
Have you tried accessing the node port under the Minishift IP? Something like `$(minishift ip): 30528`. I am not 100% sure whether this is going to work. The easier is to use _oc port-forward_.

Something like:

    $ oc port-froward POD [LOCAL_PORT:]REMOTE_PORT

Does that work for you?

**Comment by hferentschik** [comment]
@durandom, I actually think connecting via VM IP + node port should work. Here is what I just tried, using MariaDB as an example:

    $ minishift start
    $ eval $(minishift oc-env)
    $ oc new-app -e MYSQL_ROOT_PASSWORD=foo https://raw.githubusercontent.com/openshift/origin/master/examples/db-templates/mariadb-persistent-template.json
    $ oc rollout status -w dc/mariadb
    $ oc expose dc mariadb --type=LoadBalancer --name=mariadb-ingress
    $ oc export svc mariadb-ingress
     ....
    ports:
        - nodePort: 30907
     ....
    $ mysql --user=root --password=foo --host=$(minishift ip) --port=30907

The part which I think is important is:

    $ oc expose dc mariadb --type=LoadBalancer --name=mariadb-ingress

See also https://docs.openshift.org/latest/dev_guide/getting_traffic_into_cluster.html#using-externalIP

I am then able to connect to the database from my host using the Minishift VM IP + node port.

**Comment by durandom** [comment]
@hferentschik awesome, that worked for my kafka template too
many thanks for your help

**Comment by durandom** [comment]
And for reference: https://github.com/durandom/manageiq-dos/tree/master/kafka is where I put info on kafka on minishift

**Comment by hferentschik** [comment]
> awesome, that worked for my kafka template too many thanks for your help

great that it worked. I will make sure we add something to the documentation. This has come up a couple of times now.

**Comment by hferentschik** [comment]
In fact, I'll reopen the issue and put the documentation label on, so that we can track this as a documentation task.

**Comment by hferentschik** [comment]
Additional docs merged via pull request #1035. Thanks @durandom for the discussion around this.
